### PR TITLE
ISPN-6010 JPA cache store uses wrong class loader to load entities

### PR DESCRIPTION
**Pull request by mgencur**
https://issues.jboss.org/browse/ISPN-6010

### CODE REVIEW

**Comment by mgencur** [comment]
Updated my PR - also change the 7.2 parser and simplify tests.

File: integrationtests/as-integration-embedded/src/test/java/org/infinispan/test/integration/as/InfinispanStoreJpaIT.java [hunk 1]
```diff
@@ -64,4 +69,21 @@ public void testCacheManager() {
...
    }
 
+   @Test
+   public void testXmlConfig() throws IOException {
+      EmbeddedCacheManager cm = new DefaultCacheManager("jpa-config.xml");
+      Cache<String, KeyValueEntity> specificCache = cm.getCache("specificCache");
+      validateConfig(specificCache);
+      KeyValueEntity entity = new KeyValueEntity("k", "v");
+      specificCache.put(entity.getK(), entity);
+      cm.stop();
```
Comment (danberindei) [inline integrationtests/as-integration-embedded/src/test/java/org/infinispan/test/integration/as/InfinispanStoreJpaIT.java]: `stop()` should be in a finally block, or you could use `TestingUtil.withCacheManager()`.

**Comment by mgencur** [comment]
Fixed.

**Comment by danberindei** [comment]
I see a failure in `as.InfinispanStoreJpaIT`:

http://ci.infinispan.org/viewLog.html?buildId=33148&tab=buildResultsDiv&buildTypeId=bt9

**Comment by mgencur** [comment]
Sorry. Forgot to commit jpa-config.xml file for the test. It is there now.

**Comment by danberindei** [comment]
Integrated, thanks Martin!

**Pull request closed without merging**